dot/sync/chain_processor_integration_test.go

```diff
@@ -13,33 +13,127 @@ import (
 	"github.com/ChainSafe/gossamer/dot/network"
 	"github.com/ChainSafe/gossamer/dot/state"
 	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/babe"
+	"github.com/ChainSafe/gossamer/lib/common"
 	"github.com/ChainSafe/gossamer/lib/common/variadic"
 	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/ChainSafe/gossamer/pkg/scale"
 
 	"github.com/stretchr/testify/require"
 )
 
-// TODO: add test against latest gssmr runtime
-// See https://github.com/ChainSafe/gossamer/issues/2703
-func TestChainProcessor_HandleBlockResponse_ValidChain(t *testing.T) {
-	syncer := newTestSyncer(t)
-	responder := newTestSyncer(t)
+func buildBlockWithSlotAndTimestamp(t *testing.T, instance state.Runtime,
+	parent *types.Header, currentSlot, timestamp uint64) *types.Block {
+	t.Helper()
 
-	// get responder to build valid chain
-	parent, err := responder.blockState.(*state.BlockState).BestBlockHeader()
+	digest := types.NewDigest()
+	prd, err := types.NewBabeSecondaryPlainPreDigest(0, currentSlot).ToPreRuntimeDigest()
+	require.NoError(t, err)
+	err = digest.Add(*prd)
 	require.NoError(t, err)
+	header := &types.Header{
+		ParentHash:     parent.Hash(),
+		StateRoot:      common.Hash{},
+		ExtrinsicsRoot: common.Hash{},
+		Number:         parent.Number + 1,
+		Digest:         digest,
+	}
 
-	bestBlockHash := responder.blockState.(*state.BlockState).BestBlockHash()
-	rt, err := responder.blockState.GetRuntime(bestBlockHash)
+	err = instance.InitializeBlock(header)
+	require.NoError(t, err)
+
+	inherentData := types.NewInherentData()
+	err = inherentData.SetInherent(types.Timstap0, timestamp)
+	require.NoError(t, err)
+
+	err = inherentData.SetInherent(types.Babeslot, currentSlot)
+	require.NoError(t, err)
+
+	parachainInherent := babe.ParachainInherentData{
+		ParentHeader: *parent,
+	}
+
+	err = inherentData.SetInherent(types.Parachn0, parachainInherent)
+	require.NoError(t, err)
+
+	err = inherentData.SetInherent(types.Newheads, []byte{0})
+	require.NoError(t, err)
+
+	encodedInherentData, err := inherentData.Encode()
+	require.NoError(t, err)
+
+	// Call BlockBuilder_inherent_extrinsics which returns the inherents as encoded extrinsics
+	encodedInherentExtrinsics, err := instance.InherentExtrinsics(encodedInherentData)
+	require.NoError(t, err)
+
+	// decode inherent extrinsics
+	var inherentExtrinsics [][]byte
+	err = scale.Unmarshal(encodedInherentExtrinsics, &inherentExtrinsics)
```

Comment:
nit unneeded comment

```suggestion
	var inherentExtrinsics [][]byte
	err = scale.Unmarshal(encodedInherentExtrinsics, &inherentExtrinsics)
```